app/scripts/controllers/network/createInfuraClient.test.js

```diff
@@ -0,0 +1,232 @@
+ */
+
+import { withInfuraClient } from './provider-api-tests/helpers';
+import {
+  testsForRpcMethodThatDoesNotSupportParams,
+  testsForRpcMethodsThatCheckForBlockHashInResponse,
+  testsForRpcMethodThatSupportsMultipleParams,
+} from './provider-api-tests/shared-tests';
+
+describe('createInfuraClient', () => {
+  // The first time an RPC method is requested, the latest block number is
+  // pulled from the block tracker, the RPC method is delegated to Infura, and
+  // the result is cached under that block number, as long as the result is
+  // "non-empty". The next time the same request takes place, Infura is not hit,
+  // and the result is pulled from the cache.
+  //
+  // For most RPC methods here, a "non-empty" result is a result that is not
+  // null, undefined, or a non-standard "nil" value that geth produces.
+  //
+  // Some RPC methods are cacheable. Consult the definitive list of cacheable
+  // RPC methods in `cacheTypeForPayload` within `eth-json-rpc-middleware`.
+
+  describe('when the RPC method is eth_chainId', () => {
+    it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a hex string', async () => {
+      const chainId = await withInfuraClient(
+        { network: 'ropsten' },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'eth_chainId',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('0x3');
+    });
+  });
+
+  describe('when the RPC method is net_version', () => {
+    it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a decimal string', async () => {
+      const chainId = await withInfuraClient(
+        { network: 'ropsten' },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'net_version',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('3');
+    });
+  });
+
+  // == RPC methods that do not support params
+  //
+  // For `eth_getTransactionByHash` and `eth_getTransactionReceipt`, a
+  // "non-empty" result is not only one that is not null, undefined, or a
+  // geth-specific "nil", but additionally a result that has a `blockHash` that
+  // is not 0x0.
+
+  describe('eth_blockNumber', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_blockNumber');
+  });
+
+  describe('eth_compileLLL', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileLLL');
+  });
+
+  describe('eth_compileSerpent', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileSerpent');
+  });
+
+  describe('eth_compileSolidity', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_compileSolidity');
+  });
+
+  describe('eth_estimateGas', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_estimateGas');
+  });
+
+  describe('eth_gasPrice', () => {
+    testsForRpcMethodThatDoesNotSupportParams('eth_gasPrice');
+  });
+
+  describe('eth_getBlockByHash', () => {
```

Comment:
This method does support parameters (see https://docs.infura.io/infura/networks/ethereum/json-rpc-methods/eth_getblockbyhash)

Reply:
Looks like this is another case where the position of the block ref has confused matters. This method does support parameters, but the blockRef is the first one.

Reply:
Similar case with `eth_getTransactionByBlockHashAndIndex` and `eth_getTransactionByBlockNumberAndIndex`

Reply:
Hopefully fixed in https://github.com/MetaMask/metamask-extension/pull/15556/commits/f815a103f8bf7077869f8119365e2d14fc50d4ed.